packages/gatsby/index.d.ts

```diff
@@ -1521,3 +1522,57 @@ export interface IPluginRefOptions {
 export interface PluginOptionsSchemaArgs {
   Joi: PluginOptionsSchemaJoi
 }
+
+/**
+ * Send body of response
+ */
+type Send<T> = (body: T) => void
+
+/**
+ * Gatsby API Function route response
+ */
+export type GatsbyAPIFunctionResponse<T = any> = ServerResponse & {
+  /**
+   * Send `any` data in response
+   */
+  send: Send<T>
+  /**
+   * Send `json` data in response
+   */
+  json: Send<T>
+  /**
+   * Set the http status code of the response
+   */
+  status: (statusCode: number) => GatsbyAPIFunctionResponse<T>
+  redirect(url: string): GatsbyAPIFunctionResponse<T>
+  redirect(status: number, url: string): GatsbyAPIFunctionResponse<T>
+}
+
+/**
+ * Gatsby API function route request
+ */
+export interface GatsbyAPIFunctionRequest extends IncomingMessage {
+  /**
+   * Object of values from URL query parameters (after the ? in the URL)
+   */
+  query: {
```

Comment:
Type these as `Record<string, string>`

Reply:
This suggests that `{ [k: string]: string }` is more idiomatic? Don't care much either way again but curious 🤷 

https://stackoverflow.com/a/51937036/182702

Reply:
changed

Reply:
I'm not sure the basis of that saying it's more idiomatic. Maybe because it's been around longer? I prefer `Record` because it's easier to read because it's analogous to `Map`. We use `Array` over `[]` for the same reason, as it's analogous to `Set`.